### PR TITLE
Solving Incorrect assembly version in NuGet package

### LINKED ISSUE #745: Incorrect assembly version in NuGet package
**Describe your environment.**
* SDK version: 0.2.0-alpha.275

**Steps to reproduce.**
Download the nuget package, extrac the binary file, use [ILSpy](https://github.com/icsharpcode/ILSpy/releases) to check the assembly version.

**What is the expected behavior?**
Assembly version should be `0.2.0.275`.

**What is the actual behavior?**
Assembly version is `0.0.0.0`.

### PR DESCRIPTION
**Pull request by eddynaka**
Closes #745.

## Changes
- updated the version of MinVer that we use from 2.0.0 to 2.3.0, since the old version contained a bug which prevented minver from updating the correct tags
- added a tag in common.props to update version manually

### Checklist
- [ ] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public API reviewed

The PR will automatically request reviews from [code owners](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/CODEOWNERS#L15) and trigger CI build and tests.

### CODE REVIEW

**Comment by eddynaka** [comment]
Before the change:
image

After the change:
image

File: build/Common.prod.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <ItemGroup>
-    <PackageReference Include="MinVer" Version="2.0.0">
+    <PackageReference Include="MinVer" Version="2.3.0">
```
Comment (cijothomas) [inline build/Common.prod.props:5]: was this the rootcause of the issue?

Comment (eddynaka) [reply build/Common.prod.props:5]: i saw that just updating it, it started to print the correct values. Still checking if i have to configure more stuff

Comment (eddynaka) [reply build/Common.prod.props:5]: I saw that the assemblyVersion is still missing, so, it's not complete yet.

Comment (eddynaka) [reply build/Common.prod.props:5]: the reason that this wasn't working correctly: https://github.com/adamralph/minver/issues/342, looks like for netcore3.1, the way it builds make it to don't run properly, so the new version started to work correctly.

Comment (eddynaka) [reply build/Common.prod.props:5]: @cijothomas , with the new commit it solves the issue with the version 0.0.0.0

**Comment by eddynaka** [comment]
> Thank! @reyang can you also review this.

@cijothomas / @reyang , if you check the action for this PR, you can download the artifact, which will contain the nupkg file

**Pull request merged**